packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemPath.kt

```diff
@@ -1,39 +1,44 @@
 package expo.modules.filesystem.next
 
+import android.net.Uri
 import android.os.Build
+import androidx.core.net.toUri
 import expo.modules.interfaces.filesystem.Permission
 import expo.modules.kotlin.sharedobjects.SharedObject
 import java.io.File
+import java.net.URI
 import java.nio.file.attribute.BasicFileAttributes
 import java.util.EnumSet
 import kotlin.io.path.Path
 import kotlin.io.path.moveTo
 import kotlin.io.path.readAttributes
 import kotlin.time.Duration.Companion.milliseconds
 
-// We use the `File` class to represent a file or a directory in the file system.
-// The Path class might be better, but `java.nio.file.Path` class is not available in API 23.
-// The URL, URI classes seem like a less suitable choice.
-// https://stackoverflow.com/questions/27845223/whats-the-difference-between-a-resource-uri-url-path-and-file-in-java
-abstract class FileSystemPath(public var file: File) : SharedObject() {
-  fun delete(fileOrDirectory: File = file) {
-    if (!fileOrDirectory.exists()) {
-      throw UnableToDeleteException("path '${fileOrDirectory.path}' does not exist")
+abstract class FileSystemPath(var uri: Uri) : SharedObject() {
+  val javaFile get() = File(URI.create(uri.toString()))
+  val file: FileSystemFileAdapter get() {
+    return FileSystemFileAdapter(appContext?.reactContext ?: throw Exception("No context"), uri)
+  }
```

Comment:
```suggestion
  val file: FileSystemFileAdapter = FileSystemFileAdapter(appContext?.reactContext ?: throw Exceptions.ReactContextLost(), uri)
```
Do we need to create a new adapter for each get?